### PR TITLE
[http-proxy] update TargetUrl

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Template

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.npmjs.com/package/http-proxy#http---https-using-a-pkcs12-client-certificate

## Changes

Reflect documented actual possibility with `target`
Documented in [Readme](https://www.npmjs.com/package/http-proxy#http---https-using-a-pkcs12-client-certificate) and [test](https://github.com/http-party/node-http-proxy/blob/a3fe02d651d05d02d0ced377c22ae8345a2435a4/test/lib-http-proxy-common-test.js#L317-L334)

### CODE REVIEW

File: types/http-proxy/index.d.ts [hunk 1]
```diff
@@ -16,8 +17,25 @@ import * as events from "events";
 import * as url from "url";
 import * as stream from "stream";
 
+type ProxyTarget = ProxyTargetUrl | ProxyTargetDetailled;
```
Comment (armanio123) [inline types/http-proxy/index.d.ts]: There's a typo on `Detailled`.

File: types/http-proxy/index.d.ts [hunk 2]
```diff
@@ -16,8 +17,25 @@ import * as events from "events";
...
+
 type ProxyTargetUrl = string | Partial<url.Url>;
 
+interface ProxyTargetDetailled {
```
Comment (armanio123) [inline types/http-proxy/index.d.ts]: Could you please point me the documentation or code that uses this extra properties. The `Partial<url.Url>` already define some of this and I didn't found the details on the documentation.

**Review by armanio123 — CHANGES_REQUESTED** [review]

File: types/http-proxy/index.d.ts (hunk 2 above)
Comment (bodinsamuel) [reply types/http-proxy/index.d.ts]: Yes it's partially covered by `Partial<url.Url>` but some of them are just slightly different, e.g: port is a string in url.URL and the lib allow number
I had put one link in the pr but it's indeed not enough:
- https://www.npmjs.com/package/http-proxy#http---https-using-a-pkcs12-client-certificate
- https://github.com/http-party/node-http-proxy/blob/a3fe02d651d05d02d0ced377c22ae8345a2435a4/lib/http-proxy/common.js#L37
- https://github.com/http-party/node-http-proxy/blob/a3fe02d651d05d02d0ced377c22ae8345a2435a4/test/lib-http-proxy-common-test.js#L322

File: types/http-proxy/index.d.ts (hunk 1 above)
Comment (bodinsamuel) [reply types/http-proxy/index.d.ts]: oh indeed 🤭

File: types/http-proxy/index.d.ts (hunk 2 above)
Comment (armanio123) [reply types/http-proxy/index.d.ts]: Thanks for pointing the code. Merging the PR.

**Review by armanio123 — APPROVED** [review]

**Comment by bodinsamuel** [comment]
thanks @armanio123 🙏

**Pull request merged**